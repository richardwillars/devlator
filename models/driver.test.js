

const chai = require('chai');

const expect = chai.expect;
const sinon = require('sinon');
const sinonChai = require('sinon-chai');

chai.use(sinonChai);
const mockery = require('mockery');

describe('models/driver', () => {
  let moduleToBeTested;
  let driverConstructorSpy;
  const modelSpy = sinon.spy();

  beforeEach((done) => {
    driverConstructorSpy = sinon.spy();

    const schemaClass = class Event {
      constructor(props) {
        driverConstructorSpy(props);
      }
    };
    const mongooseMock = {
      Schema: schemaClass,
      model: (schemaId, schema) => {
        modelSpy(schemaId, schema);
        return schema;
      },
    };


    mockery.enable({
      useCleanCache: true,
      warnOnUnregistered: false,
    });

    mockery.registerMock('mongoose', mongooseMock);
    done();
  });

  afterEach((done) => {
    mockery.deregisterMock('mongoose');
    done();
  });

  it('should create a mongoose schema representing a driver', () => {
    // call the module to be tested

    moduleToBeTested = require('../../models/driver');
    expect(driverConstructorSpy).to.have.been.calledOnce;
    expect(driverConstructorSpy).to.have.been.calledWith({
      _id: {
        type: String,
        required: true,
        unique: true,
      },
      settings: {
        type: Object,
        required: true,
        default: {},
      },
    });
  });

  it('should create a mongoose model from the schema', () => {
    moduleToBeTested = require('../../models/driver');
    expect(modelSpy).have.been.calledOnce;
    expect(modelSpy).to.have.been.calledWith('Driver');
    expect(moduleToBeTested.Model).to.be.an.object;
  });
});
