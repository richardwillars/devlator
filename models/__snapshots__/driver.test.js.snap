// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/driver should export the model 1`] = `
Object {
  "_id": Object {
    "required": true,
    "type": [Function],
    "unique": true,
  },
  "settings": Object {
    "default": Object {},
    "required": true,
    "type": [Function],
  },
}
`;

exports[`models/driver should export the schema 1`] = `
Object {
  "_id": Object {
    "required": true,
    "type": [Function],
    "unique": true,
  },
  "settings": Object {
    "default": Object {},
    "required": true,
    "type": [Function],
  },
}
`;
